### ISSUE TITLE
[DevTools / JS / Get Started / (3) Paused On Correct Line]

### ISSUE THREAD

**Issue by rlevine** [google/WebFundamentals#4460]
*snort*

The debug demo is running minified js, so it's hard to follow along with the instructions to look for specific variables and set watches on them...

:)

But it's a nice tutorial - thanks for all the effort that went into it!

**Comment by kaycebasques** [comment]
Hey @rlevine, the demo isn't minified. My guess is that you have an extension that intercepts all `click` events. So you're actually paused in the extension's handler. If you try running the demo again in Incognito Mode you probably won't be able to reproduce what you're seeing. This comes up often enough that I covered it in the troubleshooting for step 2 (i.e. when you click the **something's wrong** button) but I don't think I've seen it get reported for step 3 too often.

Let me know if that helps

Glad to hear you're liking the tutorial. It took me a lot of effort

**Comment by rlevine** [comment]
Hi Caye!

Yep, that was the issue.

Here’s a bit of sequence information, if it’s useful. (Full disclosure, I’ve taught debugging and tools for other languages a bunch, but haven’t spent much time in Chrome, so I went in knowing more than some in your audience. I was as curious about the demo as I was the teaching content. :)

I’m running 1password, and it does intercept click events. 

When opened the inspector for the first time and clicked on “Sources,” I saw tabs for both the demo html and js files in the tab header, but also for the 1password lib, end.min.js. The file get-started (with no .html extension shown) was selected in the left nav column, so I didn’t think twice about the code display. I then set a mouse click breakpoint, returned to your demo window, and clicked the demo button. The click triggered the breakpoint, and I hit your Step 2 “I TRIGGERED THE BREAKPOINT” feedback button, ‘cause I did trigger it.



At this point, I read Step 3, and I was on line 52, not 19. Bzzzt. 

I got this from the “I’M PAUSED ON ANOTHER LINE” troubleshooting button:

 

Looking at the troubleshooting feedback for Step 2, it seems like it’s swapped with troubleshooting for step 3:



In any case, as I said in my first note, it’s a cool demo. Even though it’s short, this kind of interactive content takes a tremendous amount of work to make it both bombproof and be perceived like it flows easily. The good ones end up being labors of love, eating up far more time than intended.

Thanks for creating it!

Rick Levine

+1 303 589-6700

***
